index.js

```diff
@@ -732,21 +848,24 @@ Peer.prototype._onChannelClose = function () {
   self.destroy()
 }
 
-Peer.prototype._onAddStream = function (event) {
-  var self = this
-  if (self.destroyed) return
-  self._debug('on add stream')
-  self.emit('stream', event.stream)
-}
-
 Peer.prototype._onTrack = function (event) {
   var self = this
   if (self.destroyed) return
+
+  self._remoteTracks.push(event.track)
   self._debug('on track')
-  var id = event.streams[0].id
-  if (self._previousStreams.indexOf(id) !== -1) return // Only fire one 'stream' event, even though there may be multiple tracks per stream
-  self._previousStreams.push(id)
-  self.emit('stream', event.streams[0])
+  self.emit('track', event.track)
```

Comment:
I'm not sure if `ontrack` is can be called multiple times (haven't found this explicitly in specification yet), but intuition indicates that it can. Hence `track` event and potentially `removetrack` might need to expose `streams` argument on them to clearly understand from which streams (if not all) track was removed.